src/ralph/models/converter.py

```diff
@@ -0,0 +1,77 @@
+"""Base converter definition"""
+
+from abc import ABC, abstractmethod
+from uuid import UUID, uuid5
+
+from pydantic import BaseModel
+
+from ralph.defaults import EDX_UUID_NAMESPACE, MODEL_PATH_SEPARATOR
+from ralph.exceptions import ConversionException
+from ralph.utils import (
+    get_dict_value_from_path,
+    remove_dict_key_from_path,
+    set_dict_value_from_path,
+)
+
+
+class BaseConverter(ABC):
+    """Converter Base Class.
+
+    Converters define a conversion set to convert from one event type to another.
+    """
+
+    __model__: BaseModel
+
+    def __init__(self):
+        """Initializes BaseConverter."""
+
+        self.conversion_dict = {}
+        self.field_array = []
+        for field in self.get_conversion_set():
+            destination_path = field[0].split(MODEL_PATH_SEPARATOR)
+            value = field[1]() if callable(field[1]) else None
+            set_dict_value_from_path(self.conversion_dict, destination_path, value)
+            if not value:
+                transformer = field[2] if len(field) == 3 else lambda x: x
+                source_path = field[1].split(MODEL_PATH_SEPARATOR)
+                self.field_array.append((destination_path, source_path, transformer))
+
+    @abstractmethod
+    def get_conversion_set(self):
+        """Returns a conversion set used for conversion."""
+
+    def convert(self, event: dict, event_str: str):
+        """Converts and serializes the event.
+
+        Args:
+            event (dict): the event to convert and serialize.
+            event_str (str): the event original string value.
+        Returns:
+            (string): converted and JSON serialized event.
```

Comment:
From an API perspective, I think we should not returned a serialized object but rather a (converted) pydantic model instance.

Reply:
Good Idea) Thanks) updated)